From b0df943eae3a3e7687dc168f6737abd02a9022db Mon Sep 17 00:00:00 2001
From: Steven Moreland <smoreland@google.com>
Date: Tue, 18 Jul 2017 16:53:53 -0700
Subject: [PATCH 4/8] nxp: jni: use proper nativehelper headers

libnativehelper exports headers under nativehelper. These were
available before incorrectly as global headers in order to give
access to jni.h.

Test: modules using packages/apps/Nfc find headers
Bug: 63762847
Change-Id: I1561141005347905c3015b9c74b322a9ceea11c1
From-Change-Id: I494e04ba7112d41f18a4b9dba9495e5f9810be22
---
 nxp/jni/com_android_nfc.cpp                         | 2 +-
 nxp/jni/com_android_nfc.h                           | 4 ++--
 nxp/jni/com_android_nfc_NativeLlcpServiceSocket.cpp | 2 +-
 nxp/jni/com_android_nfc_NativeNfcManager.cpp        | 2 +-
 nxp/jni/com_android_nfc_NativeNfcTag.cpp            | 4 ++--
 nxp/jni/com_android_nfc_NativeP2pDevice.cpp         | 2 +-
 6 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/nxp/jni/com_android_nfc.cpp b/nxp/jni/com_android_nfc.cpp
index 6f8ee1f1..72d9e691 100644
--- a/nxp/jni/com_android_nfc.cpp
+++ b/nxp/jni/com_android_nfc.cpp
@@ -20,7 +20,7 @@
 #include "com_android_nfc.h"
 #include "com_android_nfc_list.h"
 #include "phLibNfcStatus.h"
-#include <ScopedLocalRef.h>
+#include <nativehelper/ScopedLocalRef.h>
 
 /*
  * JNI Initialization
diff --git a/nxp/jni/com_android_nfc.h b/nxp/jni/com_android_nfc.h
index 67fa5815..5e2ffec0 100644
--- a/nxp/jni/com_android_nfc.h
+++ b/nxp/jni/com_android_nfc.h
@@ -27,9 +27,9 @@
 #include <log/log.h>
 #include <cutils/properties.h> // for property_get
 
-#include <JNIHelp.h>
+#include <nativehelper/JNIHelp.h>
 #include <jni.h>
-#include <ScopedLocalRef.h>
+#include <nativehelper/ScopedLocalRef.h>
 
 extern "C" {
 
diff --git a/nxp/jni/com_android_nfc_NativeLlcpServiceSocket.cpp b/nxp/jni/com_android_nfc_NativeLlcpServiceSocket.cpp
index eeeb9094..a2f57978 100644
--- a/nxp/jni/com_android_nfc_NativeLlcpServiceSocket.cpp
+++ b/nxp/jni/com_android_nfc_NativeLlcpServiceSocket.cpp
@@ -17,7 +17,7 @@
 #include <errno.h>
 #include <malloc.h>
 #include <semaphore.h>
-#include <ScopedLocalRef.h>
+#include <nativehelper/ScopedLocalRef.h>
 
 #include "com_android_nfc.h"
 
diff --git a/nxp/jni/com_android_nfc_NativeNfcManager.cpp b/nxp/jni/com_android_nfc_NativeNfcManager.cpp
index 46972f1d..f611f3e5 100644
--- a/nxp/jni/com_android_nfc_NativeNfcManager.cpp
+++ b/nxp/jni/com_android_nfc_NativeNfcManager.cpp
@@ -24,7 +24,7 @@
 #include <hardware/hardware.h>
 #include <hardware/nfc.h>
 #include <cutils/properties.h>
-#include <ScopedLocalRef.h>
+#include <nativehelper/ScopedLocalRef.h>
 
 #include "com_android_nfc.h"
 
diff --git a/nxp/jni/com_android_nfc_NativeNfcTag.cpp b/nxp/jni/com_android_nfc_NativeNfcTag.cpp
index 09af46f6..932b8410 100644
--- a/nxp/jni/com_android_nfc_NativeNfcTag.cpp
+++ b/nxp/jni/com_android_nfc_NativeNfcTag.cpp
@@ -17,8 +17,8 @@
 #include <errno.h>
 #include <malloc.h>
 #include <semaphore.h>
-#include <ScopedLocalRef.h>
-#include <ScopedPrimitiveArray.h>
+#include <nativehelper/ScopedLocalRef.h>
+#include <nativehelper/ScopedPrimitiveArray.h>
 
 #include "com_android_nfc.h"
 #include "phNfcHalTypes.h"
diff --git a/nxp/jni/com_android_nfc_NativeP2pDevice.cpp b/nxp/jni/com_android_nfc_NativeP2pDevice.cpp
index f78f6261..28bd8c00 100644
--- a/nxp/jni/com_android_nfc_NativeP2pDevice.cpp
+++ b/nxp/jni/com_android_nfc_NativeP2pDevice.cpp
@@ -18,7 +18,7 @@
 #include <errno.h>
 #include <malloc.h>
 #include <semaphore.h>
-#include <ScopedLocalRef.h>
+#include <nativehelper/ScopedLocalRef.h>
 
 #include "com_android_nfc.h"
 
-- 
2.17.1

